### PR TITLE
Feature/bsod

### PR DESCRIPTION
**Pull request by YarinOmesi**
Add class `BlueScreenOfDeath` that enbale you to trigger the bsod

### CODE REVIEW

File: BlueScreenOfDeath.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        NTSTATUS ntstatus = NtRaiseHardError(STATUS_ASSERTION_FAILURE,
+            0,
+            0,
+            IntPtr.Zero,
+            6,
+            out uint oul
+        );
+
+        return ntstatus == 0;
```
Comment (alloncm) [inline BlueScreenOfDeath.cs]: Does this code ever get executed?

I think this function should never return if the compile time flag is set

File: BlueScreenOfDeath.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    [DllImport("ntdll.dll")]
+    private static extern uint RtlAdjustPrivilege(
+        int privilege,
+        bool bEnablePrivilege,
+        bool isThreadPrivilege,
+        out bool previousValue
+    );
+
+    [DllImport("ntdll.dll")]
+    private static extern NTSTATUS NtRaiseHardError(
```
Comment (alloncm) [inline BlueScreenOfDeath.cs]: I guess those functions are not importable with the winapi lib

Comment (YarinOmesi) [reply BlueScreenOfDeath.cs]: Yes, i have tried.

File: BlueScreenOfDeath.cs (hunk 1 above)
Comment (alloncm) [reply BlueScreenOfDeath.cs]: You can add an exception with the ntstatus value in case it failed

Comment (YarinOmesi) [reply BlueScreenOfDeath.cs]: fixed

**Pull request merged**